doc/04-Upgrading.md

```diff
@@ -1,7 +1,44 @@
 # Upgrading Icinga DB
 
-Specific version upgrades are described below. Please note that version upgrades are incremental.
-If you are upgrading across multiple versions, make sure to follow the steps for each of them.
+Some Icinga DB upgrades require manual intervention, others do not. If you need to intervene, the release notes will
+point you to the specific upgrade section on this page.
+
+Please note that version upgrades are incremental. If you are upgrading across multiple versions, make sure to follow
+the steps for each of them.
+
+## Database Schema Upgrades
+
+Certain Icinga DB version upgrades require a database schema upgrade. If the upgrade section of the specific Icinga DB
+release mentions a schema upgrade, this section will guide you through the process of applying the schema upgrade.
+
+First, stop the Icinga DB daemon. If you have an HA setup, stop all Icinga DB instances.
+
+```
+systemctl stop icingadb
+```
+
+Locate the required schema upgrade files in `/usr/share/icingadb/schema/mysql/upgrades/` for MySQL/MariaDB or in
+`/usr/share/icingadb/schema/pgsql/upgrades/` for PostgreSQL. The schema upgrade files are named after the new Icinga DB
+release and are mentioned in the specific section below. If you have skipped multiple Icinga DB releases, apply all
+schema versions in their order, starting with the earliest release.
+
+The following commands would apply the 1.2.1 schema upgrade to the `icingadb` database as the `icingadb` user. Please
+modify them for your setup and the schema upgrade you want to apply.
+
+* MySQL/MariaDB:
+  ```
+  mysql -u icingadb -p icingadb < /usr/share/icingadb/schema/mysql/upgrades/1.2.1.sql
+  ```
+* PostgreSQL:
+  ```
+  psql -U icingadb icingadb < /usr/share/icingadb/schema/pgsql/upgrades/1.2.1.sql
+  ```
```

Comment:
I'd suggest avoiding migration files that actually exist in the examples. Otherwise, copy & pasting could lead to accidents one might want to avoid. Imagine you accidentally also copy the trailing `\n` and you're using a shell that doesn't detect this as a paste but simply executes, then it'll be better if you get a "no such file" error instead of reapplying a random old schema upgrade.

Given that 1.3.0 was released, 1.2.3 would be a good placeholder version that also won't exist in the future (or at least wouldn't get a schema upgrade should it ever come into existence).

Reply:
That's a valid concern, thanks. I have changed it to the non-existent version 1.2.3 and slightly rephrased the sentence above. Please take another look.